Avalonia.ListBoxAnimation.Samples/SelectingItemsControlExtension.cs

```diff
@@ -43,74 +42,153 @@ private static void SelectingItemsControlPropertyChanged(object? sender, Avaloni
-        Vector3 selectionOffset = oldSelectionVisual.Offset - newSelectionVisual.Offset;
-        // Check whether the offset is vertical (e.g. ListBox) or horizontal (e.g. TabControl)
-        // Note this code assumes the items are aligned in the SelectingItemsControl
-        bool isVerticalOffset = selectionOffset.Y != 0;
-        float offset = isVerticalOffset ? selectionOffset.Y : selectionOffset.X;
-
-        Compositor compositor = pipeVisual.Compositor;
-        // This is required
-        var quadraticEaseIn = new QuadraticEaseInOut();
-
-        // Create new offset animation between old selection position to the current position
-        Vector3KeyFrameAnimation offsetAnimation = compositor.CreateVector3KeyFrameAnimation();
-        offsetAnimation.Target = "Offset";
-        string expression = (offset > 0 ? "+" : "-") + Math.Abs(offset);
-        offsetAnimation.InsertExpressionKeyFrame(0f,
-            isVerticalOffset
-                ? $"Vector3(this.FinalValue.X, this.FinalValue.Y{expression}, 0)"
-                : $"Vector3(this.FinalValue.X{expression}, this.FinalValue.Y, 0)"
-        );
-        offsetAnimation.InsertExpressionKeyFrame(1f, "this.FinalValue");
-        offsetAnimation.Duration = TimeSpan.FromMilliseconds(250);
-
-        // Create small scale animation so the pipe will "stretch" while it's moving
-        Vector3KeyFrameAnimation scaleAnimation = compositor.CreateVector3KeyFrameAnimation();
-        scaleAnimation.Target = "Scale";
-        scaleAnimation.InsertKeyFrame(0f, Vector3.One, quadraticEaseIn);
-        scaleAnimation.InsertKeyFrame(0.5f,
-            new Vector3(1f + (!isVerticalOffset ? 0.75f : 0f), 1f + (isVerticalOffset ? 0.75f : 0f), 1f),
-            quadraticEaseIn);
-        scaleAnimation.InsertKeyFrame(1f, Vector3.One, quadraticEaseIn);
-        scaleAnimation.Duration = TimeSpan.FromMilliseconds(250);
-
-        CompositionAnimationGroup compositionAnimationGroup = compositor.CreateAnimationGroup();
-        compositionAnimationGroup.Add(offsetAnimation);
-        compositionAnimationGroup.Add(scaleAnimation);
-        ImplicitAnimationCollection pipeVisualImplicitAnimations = compositor.CreateImplicitAnimationCollection();
-        float currentOffset = isVerticalOffset ? pipeVisual.Offset.Y : pipeVisual.Offset.X;
-        if (currentOffset == 0) // Visual first shown, offset not calculated, lets trigger using Offset
-            pipeVisualImplicitAnimations["Offset"] = compositionAnimationGroup;
-        else // Visual already shown, we can't trigger on Offset as it won't change
-            pipeVisualImplicitAnimations["Visible"] = compositionAnimationGroup;
-
-        pipeVisual.ImplicitAnimations = pipeVisualImplicitAnimations;
+        }
+
+        var isVertical = deltaPos is { X: 0 };
+        var dist = (float)Math.Abs(deltaPos.X + (float)deltaPos.Y);
+        var prevSize = prevInd.Bounds.Size;
+        var nextSize = nextInd.Bounds.Size;
```

Comment:
To make sure this work on first time with virtualization (latest master) you need to fallback to the previous indicator size, e.g. `nextSize = nextInd.Bounds.Size.IsDefault ? prevSize: nextInd.Bounds.Size`
Otherwise it will not show the animation